src/main/drivers/accgyro/accgyro_bus_mpu6000.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Authors:
+ * Dominic Clifton - Cleanflight implementation
+ * John Ihlein - Initial FF32 code
+ * Konstantin Sharlaimov - busDevice refactoring
+*/
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+#include "build/debug.h"
+
+#include "common/axis.h"
+#include "common/maths.h"
+
+#include "drivers/system.h"
+#include "drivers/time.h"
+#include "drivers/io.h"
+#include "drivers/exti.h"
+#include "drivers/bus.h"
+
+#include "drivers/gyro_sync.h"
+
+#include "drivers/sensor.h"
+#include "drivers/accgyro/accgyro.h"
+#include "drivers/accgyro/accgyro_mpu.h"
+
+#if defined(USE_GYRO_SPI_MPU6000) || defined(USE_ACC_SPI_MPU6000)
+#include "drivers/accgyro/accgyro_bus_mpu6000.h"
+
+// Bits
+#define BIT_H_RESET                 0x80
+#define MPU_CLK_SEL_PLLGYROX        0x01
+#define MPU_CLK_SEL_PLLGYROZ        0x03
+#define BIT_I2C_IF_DIS              0x10
+#define BIT_GYRO                    3
+#define BIT_ACC                     2
+#define BIT_TEMP                    1
+
+// Product ID Description for MPU6000
+// high 4 bits low 4 bits
+// Product Name Product Revision
+#define MPU6000ES_REV_C4 0x14
+#define MPU6000ES_REV_C5 0x15
+#define MPU6000ES_REV_D6 0x16
+#define MPU6000ES_REV_D7 0x17
+#define MPU6000ES_REV_D8 0x18
+#define MPU6000_REV_C4 0x54
+#define MPU6000_REV_C5 0x55
+#define MPU6000_REV_D6 0x56
+#define MPU6000_REV_D7 0x57
+#define MPU6000_REV_D8 0x58
+#define MPU6000_REV_D9 0x59
+#define MPU6000_REV_D10 0x5A
+
+static bool mpu6000InitDone = false;
+
+static void mpu6000AccAndGyroInit(gyroDev_t *gyro)
+{
+    mpuIntExtiInit(gyro);
+
+    spiSetSpeed(MPU6000_SPI_INSTANCE, SPI_CLOCK_INITIALIZATON);
+
+    if (!mpu6000InitDone) {
+        // Device Reset
+        busWrite(gyro->dev, MPU_RA_PWR_MGMT_1, BIT_H_RESET);
```

Comment:
Should be `busWrite(&gyro->bus,...)` see earlier comment. `busWrite` writes to a bus, not to a dev.

Reply:
Then I'll need to refactor this. It's actually writing to a specific dev on the bus.